### PR TITLE
Form customizations

### LINKED ISSUE #5245: [Forms] improvements
Need some improvements to do an inline form with only a submit button

* dismiss inline (non-dialog) form properly onSave
* suppress required fields text if no required fields in form
* allow suppression of cancel button (for persistent forms)

### PR DESCRIPTION
**Pull request by davetsay**
Closes #5245

### Describe your changes:
* dismiss inline (non-dialog) form properly onSave
* suppress required fields text if no required fields in form
* allow suppression of cancel button (for persistent forms)

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Commit messages meet standards?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [x] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by davetsay** [comment]
The form i used was inline and the onSave button acted as a file download button. I required the above changes and some other custom hacks to achieve the desired form behavior.

Customizations needed in my form not covered by this PR
Button should act as file download
- in form onSave `.then` block created `<a>` and programmatically set url and clicked.

Form should be persistent
- in form onSave `.then` block make another call to `showForm`.

Form should not be cleared
- did not implement

File: src/api/forms/FormsAPI.js [hunk 1]
```diff
@@ -108,8 +108,8 @@ export default class FormsAPI {
         let onSave;
 
         const promise = new Promise((resolve, reject) => {
-            onSave = onFormSave(resolve);
-            onDismiss = onFormDismiss(reject);
+            onSave = onFormAction(resolve);
```
Comment (nikhilmandlik) [inline src/api/forms/FormsAPI.js:111]: 👍

**Review by nikhilmandlik — APPROVED** [review]
LGTM

**Comment by davetsay** [comment]
@jvigliotta nikhil had approved this but its been awhile so can you quickly re-review. this is the forms changes we discussed for that inline form jpl uses.

File: src/api/forms/components/FormProperties.vue [hunk 2]
```diff
@@ -24,7 +24,10 @@
 <div class="c-form js-form">
     <div class="c-overlay__top-bar c-form__top-bar">
         <div class="c-overlay__dialog-title js-form-title">{{ model.title }}</div>
-        <div class="c-overlay__dialog-hint hint">All fields marked <span class="req icon-asterisk"></span> are required.</div>
+        <div
+            v-if="hasRequiredFields"
+            class="c-overlay__dialog-hint hint"
+        >All fields marked <span class="req icon-asterisk"></span> are required.</div>
```
Comment (jvigliotta) [inline src/api/forms/components/FormProperties.vue:30]: Makes sense to me!

File: src/api/forms/components/FormProperties.vue [hunk 3]
```diff
@@ -132,6 +140,11 @@ export default {
             }
 
             return 'Cancel';
+        },
+        shouldHideCancelButton() {
+            return this.model.buttons
+                && this.model.buttons.cancel
+                && this.model.buttons.cancel.hide === true;
```
Comment (jvigliotta) [inline src/api/forms/components/FormProperties.vue:147]: Might be good to use optional chaining here? `this.model?.buttons?.cancel?.hide`

File: src/api/forms/FormsAPI.js [hunk 4]
```diff
@@ -162,26 +162,16 @@ export default class FormsAPI extends EventEmitter {
...
-
-        function onFormSave(save) {
-            return () => {
-                overlay.dismiss();
-
-                if (save) {
-                    save(changes);
+                if (callback) {
+                    callback();
                 }
```
Comment (jvigliotta) [inline src/api/forms/FormsAPI.js:175]: Is saving not a default? Do we need to pass in a save callback anywhere now?

**Review by jvigliotta — CHANGES_REQUESTED** [review]
Small change and a question.

File: src/api/forms/components/FormProperties.vue (hunk 3 above)
Comment (davetsay) [reply src/api/forms/components/FormProperties.vue:147]: 👍

File: src/api/forms/FormsAPI.js (hunk 4 above)
Comment (davetsay) [reply src/api/forms/FormsAPI.js:175]: The `onFormAction` handles both save and cancel cases, and dismisses the modal or removes the form element appropriately.

**Review by jvigliotta — APPROVED** [review]
LGTM!

**Pull request merged**